tests/TypeInfer.singletons.test.cpp

```diff
@@ -151,6 +151,24 @@ TEST_CASE_FIXTURE(Fixture, "overloaded_function_call_with_singletons")
     LUAU_REQUIRE_NO_ERRORS(result);
 }
 
+TEST_CASE_FIXTURE(Fixture, "overloaded_function_resolution_singleton_parameters")
+{
+    CheckResult result = check(R"(
```

Comment:
```suggestion
    ScopedFastFlag sff{FFlag::LuauPropagateExpectedTypesForCalls, true};

    CheckResult result = check(R"(
```

and you'll need `LUAU_FASTFLAG(LuauPropagateExpectedTypesForCalls)` in the top section of this file.